### PR TITLE
d/codeartifact_auth_token - new data source 

### LINKED ISSUE #13714: Add AWS CodeArtifact service support
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

AWS has a new service called [AWS CodeArtifact](https://aws.amazon.com/blogs/aws/software-package-management-with-aws-codeartifact/). It would be nice to get Terraform support for this.

### New or Affected Resource(s)

* aws_codeartifact_repository
* aws_codeartifact_domain

### PR DESCRIPTION
**Pull request by DrFaust92**
### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request

Closes #13714

Release note for [CHANGELOG](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md):

```release-note
datasource_codeartifact_authorization_token - new data source
```

Output from acceptance testing:

```
$ make testacc TESTARGS='-run=TestAccXXX'

...
```

### CODE REVIEW

File: website/docs/d/codeartifact_authorization_token.html.markdown [hunk 1]
```diff
@@ -0,0 +1,35 @@
+---
+subcategory: "CodeArtifact"
+layout: "aws"
+page_title: "AWS: aws_codeartifact_authorization_token"
+description: |-
+    Provides details about an CodeArtifact Authorization Token
```
Comment (ewbankkit) [inline website/docs/d/codeartifact_authorization_token.html.markdown:6]:
```suggestion
    Provides details about a CodeArtifact Authorization Token
```

File: website/docs/d/codeartifact_authorization_token.html.markdown [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+---
+
+# Data Source: aws_codeartifact_authorization_token
+
+The CodeArtifact Authorization Token data source allows the authorization token a CodeArtifact domain.
```
Comment (ewbankkit) [inline website/docs/d/codeartifact_authorization_token.html.markdown:11]:
```suggestion
The CodeArtifact Authorization Token data source generates a temporary authentication token for accessing repositories in a CodeArtifact domain.
```

**Review by ewbankkit — CHANGES_REQUESTED** [review]
Verified acceptance tests:

```console
$ make testacc TEST=./aws TESTARGS='-run=TestAccAWSCodeArtifactAuthorizationTokenDataSource_'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws -v -count 1 -parallel 20 -run=TestAccAWSCodeArtifactAuthorizationTokenDataSource_ -timeout 120m
=== RUN   TestAccAWSCodeArtifactAuthorizationTokenDataSource_basic
--- PASS: TestAccAWSCodeArtifactAuthorizationTokenDataSource_basic (18.46s)
=== RUN   TestAccAWSCodeArtifactAuthorizationTokenDataSource_owner
--- PASS: TestAccAWSCodeArtifactAuthorizationTokenDataSource_owner (18.16s)
=== RUN   TestAccAWSCodeArtifactAuthorizationTokenDataSource_duration
--- PASS: TestAccAWSCodeArtifactAuthorizationTokenDataSource_duration (19.22s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	55.888s
```

File: website/docs/d/codeartifact_authorization_token.html.markdown [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+* `domain` - (Required) The name of the domain that is in scope for the generated authorization token.
+* `domain_owner` - (Optional) The account number of the AWS account that owns the domain.
+* `duration_seconds` - (Optional) The time, in seconds, that the generated authorization token is valid. Valid values are `0` and between `900` and `43200`.
+
+## Attributes Reference
+
+In addition to the argument above, the following attributes are exported:
+
+* `authorization_token` - Temporary authorization token.
+* `domain_owner` - The account number of the AWS account that owns the domain.
```
[the remark this answers is not in the document]
Comment (DrFaust92) [reply website/docs/d/codeartifact_authorization_token.html.markdown]: done

**Review by ewbankkit — APPROVED** [review]
LGTM.

**Comment by breathingdust** [comment]
LGTM 🚀 Thanks @DrFaust92 

Verified Acceptance Tests in Commercial

```console
make testacc TEST=./aws TESTARGS='-run=TestAccAWSCodeArtifactAuthorizationTokenDataSource_'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws -v -count 1 -parallel 20 -run=TestAccAWSCodeArtifactAuthorizationTokenDataSource_ -timeout 120m
=== RUN   TestAccAWSCodeArtifactAuthorizationTokenDataSource_basic
--- PASS: TestAccAWSCodeArtifactAuthorizationTokenDataSource_basic (20.94s)
=== RUN   TestAccAWSCodeArtifactAuthorizationTokenDataSource_owner
--- PASS: TestAccAWSCodeArtifactAuthorizationTokenDataSource_owner (18.68s)
=== RUN   TestAccAWSCodeArtifactAuthorizationTokenDataSource_duration
--- PASS: TestAccAWSCodeArtifactAuthorizationTokenDataSource_duration (15.76s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	56.781s
```

**Pull request merged**